secure/json-rpc/src/lib.rs

```diff
@@ -0,0 +1,595 @@
+        ledger_version: u64,
+    ) -> Result<Response, Error> {
+        let method = "get_account_state_with_proof".to_string();
+        let params = vec![
+            Value::String(account.to_string()),
+            Value::Number(version.into()),
+            Value::Number(ledger_version.into()),
+        ];
+        Ok(self.execute_request(method, params))
+    }
+}
+
+/// Below is a sample response from a successful submit() JSON RPC call:
+/// "{
+///   "id": 0,
+///   "jsonrpc": "2.0",
+///   "result": null
+/// }"
+///
+///
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct SubmitTransactionResponse {
+    id: u64,
+    jsonrpc: String,
+    result: Option<Value>,
+}
+
+/// Below is a sample response from a successful get_account_state_with_proof_call() JSON RPC call.
+/// "{
+///   "id": 0,
+///   "jsonrpc": "2.0",
+///   "result": {
+///     "blob": "0100...",
+///     "proof": {
+///       "ledger_info_to_transaction_info_proof": "00..",
+///       "transaction_info": "200000000000000000...<truncated>...ffffffffffffffff",
+///       "transaction_info_to_account_proof": "0000..."
+///     },
+///     "version": 0
+///   }
+/// }"
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct AccountStateWithProofResponse {
+    id: u64,
+    jsonrpc: String,
+    result: AccountStateResponse,
+}
+
+/// In practice this represents an AccountStateWithProof, however, we only decode the relevant
+/// fields here.
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct AccountStateResponse {
+    version: u64,
+    blob: Option<Bytes>,
+}
+
+/// Below is a sample response from a failed JSON RPC call:
+/// "{
+///   "error": {
+///     "code": -32000,
+///     "data": null,
+///     "message": "Server error: send failed because receiver is gone"
+///   },
+///   "id": 0,
+///   "jsonrpc: "2.0"
+/// }"
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct JSONRpcFailureResponse {
+    id: u64,
+    jsonrpc: String,
+    error: JsonRpcError,
+}
+
+/// A custom error returned by the JSON RPC server for API calls that fail internally (e.g.,
+/// an internal error during execution on the server side).
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct JsonRpcError {
+    code: i16,
+    message: String,
+    data: Option<Value>,
+}
+
+/// Represents a vector of bytes using hex encoding.
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct Bytes(pub String);
+
+impl Bytes {
+    pub fn into_bytes(self) -> Result<Vec<u8>, Error> {
+        Ok(hex::decode(self.0)?)
+    }
+}
+
+impl From<&[u8]> for Bytes {
+    fn from(bytes: &[u8]) -> Self {
+        Self(hex::encode(bytes))
+    }
+}
+
+impl From<&Vec<u8>> for Bytes {
+    fn from(bytes: &Vec<u8>) -> Self {
+        Self(hex::encode(bytes))
+    }
+}
+
+/// Testing infrastructure for the lightweight JSON client above. Hooks into the original JSON RPC
+/// infrastructure.
+#[cfg(test)]
+mod test {
+    use crate::JsonRpcClient;
```

Comment:
always recommend using super::* first and then pulling in additional dependencies that aren't in super::*
this is the only time I /ever/ recommend using * :P